### PR TITLE
DCS-1470 set state for no match journey and fixed no match redirect o…

### PR DESCRIPTION
**Pull request by louisenorris**
…n confirmation page

### CODE REVIEW

File: server/routes/bookedtoday/arrivals/checkAnswersController.ts [hunk 1]
```diff
@@ -61,7 +61,7 @@ export default class CheckAnswersController {
       req.flash('arrivalResponse', {
         firstName: arrival.firstName,
         lastName: arrival.lastName,
-        prisonNumber: arrival.prisonNumber,
+        prisonNumber: arrivalResponse.prisonNumber,
```
Comment (louisenorris) [inline server/routes/bookedtoday/arrivals/checkAnswersController.ts:64]: changed to obtain prisonNumber from arrivalResponse as this no match journey has an undefined prisonNumber in arrival object which was causing a redirect to the choose prisoner page in the confirmation controller as prisonNumber was undefined in flash.

Comment (andrewrlee) [reply server/routes/bookedtoday/arrivals/checkAnswersController.ts:64]: yep that makes sense

File: server/routes/bookedtoday/arrivals/noRecordFoundController.ts [hunk 2]
```diff
@@ -8,6 +10,16 @@ export default class NoRecordFoundController {
...
       const data = await this.expectedArrivalsService.getArrival(id)
+
+      State.newArrival.set(res, {
+        firstName: convertToTitleCase(data.firstName),
+        lastName: convertToTitleCase(data.lastName),
+        dateOfBirth: data.dateOfBirth,
+        sex: data.gender,
+        prisonNumber: data.prisonNumber,
+        pncNumber: data.pncNumber,
+      })
```
Comment (andrewrlee) [inline server/routes/bookedtoday/arrivals/noRecordFoundController.ts:21]: this should be done in a post handler - same as the single match one should be moved to

**Review by andrewrlee — APPROVED** [review]
LGTM

File: server/routes/bookedtoday/arrivals/noRecordFoundController.ts (hunk 2 above)
Comment (andrewrlee) [reply server/routes/bookedtoday/arrivals/noRecordFoundController.ts:21]: though I guess this will move to the next page so NBD

**Pull request merged**